test-framework/sudo-compliance-tests/src/sudoedit.rs

```diff
@@ -0,0 +1,432 @@
+        .output(&env);
+
+    output.assert_exit_code(11);
+
+    let actual = Command::new("cat").arg(ETC_SUDOERS).output(&env).stdout();
+
+    assert_eq!(expected, actual);
+}
+
+#[test]
+fn temporary_file_is_deleted_during_edition() {
+    let env = Env(SUDOERS_ALL_ALL_NOPASSWD)
+        .user(USERNAME)
+        .file(
+            DEFAULT_EDITOR,
+            TextFile(
+                "#!/bin/sh
+rm $1",
+            )
+            .chmod(CHMOD_EXEC),
+        )
+        .build();
+
+    let output = Command::new("sudoedit")
+        .arg(ETC_SUDOERS)
+        .as_user(USERNAME)
+        .output(&env);
+
+    output.assert_exit_code(1);
+    let stderr = output.stderr();
+    if sudo_test::is_original_sudo() {
+        assert_contains!(stderr, format!("sudoedit: {ETC_SUDOERS} left unmodified"));
+    } else {
+        assert_contains!(stderr, format!("Failed to read from temporary file"));
+    }
+}
+
+#[test]
+fn temp_file_initial_contents() {
+    let expected = "
+Defaults !fqdn
+ALL ALL=(ALL:ALL) NOPASSWD:ALL";
+    let env = EnvNoImplicit(expected)
+        .user(USERNAME)
+        .file(
+            DEFAULT_EDITOR,
+            TextFile(format!(
+                "#!/bin/sh
+cp $1 {LOGS_PATH}"
+            ))
+            .chmod(CHMOD_EXEC),
+        )
+        .build();
+
+    Command::new("sudoedit")
+        .arg(ETC_SUDOERS)
+        .as_user(USERNAME)
+        .output(&env)
+        .assert_success();
+
+    let actual = Command::new("cat").arg(LOGS_PATH).output(&env).stdout();
+
+    assert_eq!(expected, actual);
+}
+
+#[test]
+fn temporary_file_is_deleted_when_done() {
+    let expected = SUDOERS_ALL_ALL_NOPASSWD;
+    let env = Env(expected)
+        .user(USERNAME)
+        .file(DEFAULT_EDITOR, TextFile(EDITOR_DUMMY).chmod(CHMOD_EXEC))
+        .build();
+
+    Command::new("sudoedit")
+        .arg("/foo.txt")
+        .as_user(USERNAME)
+        .output(&env)
+        .assert_success();
+
+    let output = Command::new("find")
+        .args(["/var/tmp", "-type", "f"])
+        .output(&env)
+        .stdout();
+
+    assert!(output.is_empty());
+}
+
+#[test]
+fn temporary_file_is_deleted_when_terminated_by_signal() {
+    for victim in ["child", "parent"] {
+        let kill_sudo = "/root/kill-sudo.sh";
+        let env = Env(SUDOERS_ALL_ALL_NOPASSWD)
+            .user(USERNAME)
+            .file(
+                DEFAULT_EDITOR,
+                TextFile(
+                    "#!/bin/sh
+touch /tmp/barrier
+sleep 2",
+                )
+                .chmod(CHMOD_EXEC),
+            )
+            .file(kill_sudo, include_str!("sudoedit/kill-sudoedit.sh"))
+            .build();
+
+        let child = Command::new("sudoedit")
+            .arg(ETC_SUDOERS)
+            .as_user(USERNAME)
+            .spawn(&env);
+
+        Command::new("sh")
+            .args([kill_sudo, victim, "-TERM"])
+            .output(&env)
+            .assert_success();
+
+        assert!(!child.wait().status().success());
```

Comment:
Maybe check for the signal instead? (We should probably add `assert_signal()` next to `assert_success()` and `assert_exit_code()`)